### ISSUE TITLE
Allow smtp configure without requiring smtp user

### ISSUE THREAD

**Issue by DevKyleS** [docusealco/docuseal#23]
Please allow an SMTP configuration to be saved without forcing a smtp user to be defined

Related code
* https://github.com/docusealco/docuseal/blob/15cbdfb20c73b039499aa4a9fbcffc84f3816b53/config/environments/production.rb#L79-L90

**Comment by AlexandrToorchyn** [comment]
@DevKyleS We still need some way to determine whether the SMTP configs via ENV should be triggered or not.
I assume your local mail server is not authenticated via the username and password - in this case can we make `SMTP_ADDRESS` ENV var mantadory to enable SMTP configs?

```ruby
if ENV['SMTP_ADDRESS''] 
   config.action_mailer.delivery_method = :smtp 
   config.action_mailer.smtp_settings = { 
     address: ENV.fetch('SMTP_ADDRESS', nil), 
     port: ENV.fetch('SMTP_PORT', 587), 
     domain: ENV.fetch('SMTP_DOMAIN', nil), 
     user_name: ENV.fetch('SMTP_USERNAME', nil), 
     password: ENV.fetch('SMTP_PASSWORD', nil), 
     authentication: ENV.fetch('SMTP_AUTHENTICATION', 'plain'), 
     enable_starttls_auto: ENV['SMTP_ENABLE_STARTTLS_AUTO'] == 'true' 
   }.compact 
 end 
```

**Comment by DevKyleS** [comment]
> @DevKyleS We still need some way to determine whether the SMTP configs via ENV should be triggered or not.
> I assume your local mail server is not authenticated via the username and password - in this case can we make `SMTP_ADDRESS` ENV var mantadory to enable SMTP configs?
> 
> ```ruby
> if ENV['SMTP_ADDRESS''] 
>    config.action_mailer.delivery_method = :smtp 
>    config.action_mailer.smtp_settings = { 
>      address: ENV.fetch('SMTP_ADDRESS', nil), 
>      port: ENV.fetch('SMTP_PORT', 587), 
>      domain: ENV.fetch('SMTP_DOMAIN', nil), 
>      user_name: ENV.fetch('SMTP_USERNAME', nil), 
>      password: ENV.fetch('SMTP_PASSWORD', nil), 
>      authentication: ENV.fetch('SMTP_AUTHENTICATION', 'plain'), 
>      enable_starttls_auto: ENV['SMTP_ENABLE_STARTTLS_AUTO'] == 'true' 
>    }.compact 
>  end 
> ```

That sounds sufficient as an address will always be defined when used.

**Comment by DevKyleS** [comment]
I just confirmed release v1.0.4 resolves this sufficiently, this can be closed, thank you.

**Comment by ovizii** [comment]
I'm sorry for this blonde question, but how is this solved? (I am running a local mail proxy on 10.10.10.10 which then safely proxies my outgoing emails)
I am running docuseal locally via docker-compose. 
I have not set any .env variables regarding email. If this is required, does anyone have a link to a sample .env file please?

I am trying to save and am told that the Username field needs to be filled in.
![image](https://github.com/docusealco/docuseal/assets/837902/5e64e720-787c-4f99-9d7d-5606c2dc7401)

**Comment by DevKyleS** [comment]
@ovizii The ENV variable names can be found in this file with the last parameter being the default value if not set.
* https://github.com/docusealco/docuseal/blob/master/config/environments/production.rb#L79-L90 

Even with mine configured, I do not see the values in the `Email` page like you've screenshotted, a bug I'm sure, but the functionality is working for me (eg, forgot password email was received).

**Comment by ovizii** [comment]
Thanks for the pointer. I found those variables, but you misunderstood, what you see in the screenshot are the values I manually introduced on that screen, I had not set any via variables.

So having found the variables you linked are these the right settings for using a SMTP server on 10.10.10.10 port 25 without any authentication?

```
SMTP_ADDRESS=
SMTP_PORT=25
SMTP_DOMAIN=10.10.10.10
SMTP_USERNAME=
SMTP_PASSWORD=
SMTP_AUTHENTICATION=none
SMTP_ENABLE_STARTTLS_AUTO=false
```

I'm also confused why I can set these SMTP settings via variables AND via the GUI. If I set them via GUI I am forced to enter a user/password. Also, if I were to do that, would the GUI settings override the env variables?
